editoast/src/views/infra/mod.rs

```diff
@@ -561,6 +565,79 @@ pub mod tests {
         assert!(Infra::delete(db_pool, cloned_infra_id).await.unwrap());
     }
 
+    #[derive(QueryableByName)]
+    struct Count {
+        #[diesel(sql_type = BigInt)]
+        nb: i64,
+    }
+
+    #[rstest]
+    async fn infra_clone(db_pool: Data<DbPool>) {
+        let app = create_test_service().await;
+        let small_infra = &small_infra(db_pool.clone()).await.model;
+        let small_infra_id = small_infra.id.unwrap();
+
+        let pg_config = PostgresConfig::default();
+        let pg_config_url = pg_config.url().expect("cannot get postgres config url");
+        let mut conn = PgConnection::establish(pg_config_url.as_str())
+            .await
+            .expect("Error while connecting DB");
+
+        let infra_cache = InfraCache::load(&mut conn, small_infra).await.unwrap();
+
+        generated_data::refresh_all(db_pool.clone(), small_infra_id, &infra_cache)
+            .await
+            .unwrap();
+
+        let errors_count =
+            sql_query("SELECT COUNT (*) as nb from infra_layer_error where infra_id = $1")
+                .bind::<BigInt, _>(small_infra_id)
+                .get_result::<Count>(&mut conn)
+                .await
+                .unwrap();
+        assert!(errors_count.nb > 1);
```

Comment:
You never check that the clone infra duplicate errors. We don't need to make a special case for errors we can include it in the rest of the tables (see my following comments).